### PR TITLE
Update date picker examples, remove unused variables and add missing tests

### LINKED ISSUE #120765: Unused variable in Switch class example
Hello, in this example https://api.flutter.dev/flutter/material/Switch-class.html#material.Switch.3, the class variable `light2` is not used, so it could be removed.
Code from the example:
```dart
class SwitchExample extends StatefulWidget {
  const SwitchExample({super.key});

  @override
  State<SwitchExample> createState() => _SwitchExampleState();
}

class _SwitchExampleState extends State<SwitchExample> {
  bool light0 = true;
  bool light1 = true;
  bool light2 = true;

  final MaterialStateProperty<Icon?> thumbIcon =
      MaterialStateProperty.resolveWith<Icon?>(
    (Set<MaterialState> states) {
      // Thumb icon when the switch is selected.
      if (states.contains(MaterialState.selected)) {
        return const Icon(Icons.check);
      }
      return const Icon(Icons.close);
    },
  );

  @override
  Widget build(BuildContext context) {
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: <Widget>[
        Switch(
          value: light0,
          onChanged: (bool value) {
            setState(() {
              light0 = value;
            });
          },
        ),
        Switch(
          thumbIcon: thumbIcon,
          value: light1,
          onChanged: (bool value) {
            setState(() {
              light1 = value;
            });
          },
        ),
      ],
    );
  }
}
```

### LINKED ISSUE #121028: Unused variable in the example of FilterChip.
While reading [the official documentation of `FilterChip`](https://api.flutter.dev/flutter/material/FilterChip-class.html), I found an unused variable in the example: `bool favorite = false;`.

I don't tell if this variable is actually needed (e.g. injection or reflection purpose in test phase), so I'm posting this as an issue rather than a pull request.

location: [examples/api/lib/material/filter_chip/filter_chip.0.dart](https://github.com/flutter/flutter/blob/9944297138845a94256f1cf37beb88ff9a8e811a/examples/api/lib/material/filter_chip/filter_chip.0.dart#L33)

### PR DESCRIPTION
**Pull request by TahaTesser**
part of https://github.com/flutter/flutter/issues/101048 (final example)
fix https://github.com/flutter/flutter/issues/121028
fix https://github.com/flutter/flutter/issues/120765

### Description 

1. Update date picker examples and add missing tests for https://github.com/flutter/flutter/issues/101481
2. Remove an unused variable in the switch example and add the missing test.
3. Remove an unused variable in the filter chip example

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/material/switch/switch.2.dart [hunk 1]
```diff
@@ -35,15 +35,15 @@ class SwitchExample extends StatefulWidget {
...
-  final MaterialStateProperty<Icon?> thumbIcon = MaterialStateProperty.resolveWith<Icon?>((Set<MaterialState> states) {
-    // Thumb icon when the switch is selected.
-    if (states.contains(MaterialState.selected)) {
-      return const Icon(Icons.check);
-    }
-    return const Icon(Icons.close);
-  },
+  final MaterialStateProperty<Icon?> thumbIcon = MaterialStateProperty.resolveWith<Icon?>(
+    (Set<MaterialState> states) {
+      // Thumb icon when the switch is selected.
```
Comment (HansMuller) [inline examples/api/lib/material/switch/switch.2.dart:41]: This comment doesn't really add anything

File: examples/api/test/material/switch/switch.2_test.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      Icons.check,
+    );
+    expect(
+      materialSwitch.thumbIcon!.resolve(unselected)!.icon,
+      Icons.close,
+    );
+  });
+}
+
+Set<MaterialState> selected = <MaterialState>{ MaterialState.selected };
```
Comment (HansMuller) [inline examples/api/test/material/switch/switch.2_test.dart:31]: I think these vars would be better off as const definitions within the test.

**Review by HansMuller — COMMENTED** [review]
Looks good, just some nit picking, mostly about the existing code.

File: examples/api/lib/material/filter_chip/filter_chip.0.dart [hunk 3]
```diff
@@ -30,49 +35,43 @@ class FilterChipExample extends StatefulWidget {
 }
 
 class _FilterChipExampleState extends State<FilterChipExample> {
-  bool favorite = false;
   final List<String> _filters = <String>[];
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply examples/api/lib/material/filter_chip/filter_chip.0.dart:38]: Agree. That's always my goal to make improvements while I am already updating the examples.

File: examples/api/lib/material/filter_chip/filter_chip.0.dart [hunk 4]
```diff
@@ -30,49 +35,45 @@ class FilterChipExample extends StatefulWidget {
...
+          Wrap(
+            spacing: 5.0,
+            children: ExerciseFilter.values.map((ExerciseFilter exercise) {
+              return FilterChip(
+                label: Text(exercise.name),
+                selected: filters.contains(exercise),
+                onSelected: (bool value) {
+                  setState(() {
+                    if (value) {
+                      if (!filters.contains(exercise)) {
```
Comment (HansMuller) [inline examples/api/lib/material/filter_chip/filter_chip.0.dart:59]: I don't think this test is really needed, since filters is a set.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**